synphot/reddening.py

```diff
@@ -74,11 +75,16 @@ def extinction_curve(self, ebv, wavelengths=None):
         elif not isinstance(ebv, numbers.Real):
             raise exceptions.SynphotError('E(B-V)={0} is invalid.'.format(ebv))
 
-        x = self._validate_wavelengths(wavelengths).value
-        y = 10 ** (-0.4 * self(x).value * ebv)
-        header = {
-            'E(B-V)': ebv,
-            'ReddeningLaw': self.meta.get('expr', 'unknown')}
+        x = self._validate_wavelengths(wavelengths)
+        header = {'E(B-V)': ebv}
+
+        # Duck-typing dust-extinction package API.
+        if HAS_DUST_EXTINCTION and hasattr(self.model, 'extinguish'):
+            y = self.model.extinguish(x, self.model.Rv * ebv)
+            header['ReddeningLaw'] = '{}'.format(self.model)
```

Comment:
Put the short version in header, not the multi-line one.
```suggestion
            header['ReddeningLaw'] = '{!r}'.format(self.model)
```